### ISSUE TITLE
Does the theory of dropout layer in keras refer to (Hinton et al., 2012)?

### ISSUE THREAD

**Issue by Imorton-zd** [fchollet/keras#1852]
The theory of dropout layer in keras is the same as "Hinton G., N. Srivastava, A. Krizhevsky, I. Sutskever, and R. Salakhutdinov. 2012. Improving Neural Networks by Preventing Co-adaptation of Feature Detectors. ResearchGate."?

**Comment by Imorton-zd** [comment]
@EderSantana @fchollet @dbonadiman Could you spend your valuable time solving this question? Thanks a lot.

**Comment by EderSantana** [comment]
yes, it does!

**Comment by shampool** [comment]
Theoretically it does, but the implementation is a little bit different.

**Comment by pasky** [comment]
(Also note that the p's in the original paper represent the probability of propagating, while the common current usage, also in Keras, is that p represents the probability of dropping.)

**Comment by Imorton-zd** [comment]
@EderSantana @shampool @pasky Thanks! By dropping a unit out, it means removing
the unit from the network, along with all its incoming and outgoing edges. However, I print the output of every layer. I find that the dimension of dropout layer output is the same as last output, which does not change. Why? 
`model = Sequential()
        model.add(LSTM(lstm_output_dims, input_shape=(1, max_features)))
        model.add(Dense(hidden_dims))
        model.add(Activation('relu'))
        model.add(Dropout(0.5))    
        model.add(Dense(1))
        model.add(Activation('sigmoid'))`

**Comment by shampool** [comment]
It drops the weights by multiplying them with a random matrix consists of 1 and 0, 1 means retain, 0 means dropout.
